Web UI - Troubleshooting Page

TheElixZammuto opened the pull request:
This PR adds the ability for Sunshine Users to close the application currently running (while stopping the Stream) and to unpair all clients currently paired by Sunshine. 
![TroubleShooting Example](https://user-images.githubusercontent.com/6505622/131258321-4d9fc540-3c12-4373-aee0-4375c628c25c.png)

loki-47-6F-64 commented on sunshine/nvhttp.cpp:
```diff
@@ -927,4 +927,9 @@ void start() {
+
+void erase_all_clients(){
+  map_id_client.clear();
+  save_state();
```
Nice and short. The perfect combination :)

loki-47-6F-64 commented on sunshine/confighttp.cpp:
```diff
@@ -502,6 +512,56 @@ void savePin(resp_https_t response, req_https_t request) {
+  });
+
+  try {
+    nvhttp::erase_all_clients();
```
Functions written by us shouldn't throw exceptions, if it does, that's a bug.
Pass errors by `int` instead.

loki-47-6F-64 commented on sunshine/confighttp.cpp:
```diff
@@ -502,6 +512,56 @@ void savePin(resp_https_t response, req_https_t request) {
+  });
+
+  try {
+    proc::proc.terminate();
```
Functions written by us shouldn't throw exceptions, if it does, that's a bug.
Pass errors by `int` instead.

loki-47-6F-64 commented on sunshine/confighttp.cpp:
```diff
@@ -502,6 +512,56 @@ void savePin(resp_https_t response, req_https_t request) {
+    outputTree.put("status", false);
+    outputTree.put("error", e.what());
+    return;
+  }
```
outputTree.put shouldn't throw an exceptions if handled right anyway.

loki-47-6F-64 commented on sunshine/confighttp.cpp:
```diff
@@ -502,6 +512,56 @@ void savePin(resp_https_t response, req_https_t request) {
+    outputTree.put("status", false);
+    outputTree.put("error", e.what());
+    return;
+  }
```
outputTree.put shouldn't throw an exceptions if handled right anyway.

loki-47-6F-64 reviewed (changes requested):
Once again, thanks for your work :)

TheElixZammuto commented:
I updated the repo to remove the handling of our method calls, the methods that I use in particular are void and should not have any exception or strange problems on their side. So I removed the check entierly. I also fixed the style of the cards
